### PR TITLE
weather Sandra Shumanteva Kaevska

### PR DESCRIPTION
**Pull request by sandra-shumanteva-kaevska**
Link to my deployed project:

My reflections on how this week's project turned out:

Things I'd like to have clarified or explained in more detail:

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    />
+    <title>Weather</title>
+  </head>
+  <body>
+    <section class="cities">
+      <div class="choose-cities">
+        <h3>You can choose to check weather in this cities:</h3>
+
+        <label for="skopje"
+          >Skopje
```
Comment (karanmann) [inline index.html:19]: I think you wanted to have the > where the tag finished.

File: index.html [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            type="radio"
+            id="paris"
+            name="city"
+            value="Paris,France"
+            onchange="getValue(this)"
+          />
+        </label>
+
+        <label for="stockholm"
+          >Stockholm
```
Comment (karanmann) [inline index.html:41]: The same here. I sorry if I am being a bit picky. Its just make the code cleaner. Its a great and well composed HTML with proper labels. Great Job :)

File: weather.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+const weatherIcon = document.getElementById("weatherIcon");
+const averageTemp = document.getElementById("averageTemp");
+const minTemp = document.getElementById("minTemp");
+const maxTemp = document.getElementById("maxTemp");
+const feelsLike = document.getElementById("feelsLike");
+const windSpeed = document.getElementById("windSpeed");
+const sunrise = document.getElementById("sunrise");
+const sunset = document.getElementById("sunset");
+const forcast = document.getElementsByClassName("forcast");
+
```
Comment (karanmann) [inline weather.js:14]: Its really nice that you have defined all the constants in one place and also added a lot of components to the JavaScript from the API

File: weather.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+      weatherIcon.src = `https://openweathermap.org/img/wn/${weatherArr.weather[0].icon}.png`;
+      averageTemp.innerHTML = generateRoundedTemperature(weatherArr.main.temp);
+      minTemp.innerHTML = generateRoundedTemperature(weatherArr.main.temp_min);
+      maxTemp.innerHTML = generateRoundedTemperature(weatherArr.main.temp_max);
+      feelsLike.innerHTML = generateRoundedTemperature(
+        weatherArr.main.feels_like
+      );
+      document.getElementById(
+        "todayWeather"
+      ).style.backgroundImage = background(weatherArr.main.temp);
```
Comment (karanmann) [inline weather.js:34]: Nice that you styled your background from JS :)

File: README.md [hunk 5]
```diff
@@ -1,13 +1,11 @@
...
 
-Replace this readme with your own information about your project.
-
-Start by briefly describing the assignment in a sentence or two. Keep it short and to the point.
+This projet is for weather forcast.
 
 ## The problem
 
-Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
+I used API to fetch the current weather and weather prognosis for the next days. I created different functions that would fetch and set the weather data in the HTML. I used google search and stackOverflow to find informations for my problems.
```
Comment (karanmann) [inline README.md:7]: I would encourage you to fill out this section with the problems you faced so you can refer back to them at a later stage :)

File: style.css [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+@media (min-width: 667px) and (max-width: 1024px) {
+  .temperature {
+    flex-direction: row;
+  }
+  .forcast {
+    flex-direction: row;
+  }
+}
```
Comment (karanmann) [inline style.css:59]: Great job with the CSS and making the website responsive :)

File: weather.js [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+const weatherToday = (cityName) => {
+  const apiWeather = `https://api.openweathermap.org/data/2.5/weather?q=${cityName}&units=metric&APPID=9dcf660f4adfbe64a1f1edf5962b352d`;
+  fetch(apiWeather)
+    .then((weather) => {
+      return weather.json();
+    })
+    .then((weatherArr) => {
+      city.innerHTML = weatherArr.name;
+      country.innerHTML = weatherArr.sys.country;
+      description.innerHTML = weatherArr.weather[0].description;
```
Comment (karanmann) [inline weather.js:24]: Its great that you took the average to get a more accurate measurement.

**Review by karanmann — APPROVED** [review]
Hey Sandra,
Firstly congratulation on achieving the Red Level. Your code was well structured and very easy to read. It clear that you have a good grasp of how JS works with HTML and also how to retrieve and mould the data from an API. It was a pleasure reading your code :)

File: index.html [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+      </section>
+
+      <section class="forcast">
+        <div class="filtredForcastDate">Date:</div>
+        <div>
+          Temperature: <span class="filtredForcastTemperatureMin"></span> °C
+        </div>
+      </section>
+    </section>
+    <script src="weather.js"></script>
```
Comment (HIPPIEKICK) [inline index.html:118]: `./weather.js`

File: style.css (hunk 6 above)
Comment (HIPPIEKICK) [reply style.css:59]: Agreed! Though - would it make more sense (when we are styling mobile first) that tablet came second and desktop last? ;)

File: weather.js [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+      weatherIcon.src = `https://openweathermap.org/img/wn/${weatherArr.weather[0].icon}.png`;
```
Comment (HIPPIEKICK) [inline weather.js:25]: Nice ⭐

File: weather.js [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+
+const background = (averageTemp) => {
+  if (averageTemp >= 11) {
+    document.getElementById("todayWeather").style.backgroundImage =
+      "linear-gradient(#f5d20d, #615305) ";
+  } else {
+    document.getElementById("todayWeather").style.backgroundImage =
+      "linear-gradient(#9be3ff, #0c2833)";
+  }
+};
```
Comment (HIPPIEKICK) [inline weather.js:59]: Consider toggling classes instead to be able to move the styling to the CSS-file

**Review by HIPPIEKICK — APPROVED** [review]
Good job Sandra! Especially nice with the option to choose between different cities 💯

**Pull request closed without merging**